File: app/src/main/java/com/example/moviemvp/movie_detail/MovieDetailsContract.java
package com.example.moviemvp.movie_detail;

import com.example.moviemvp.model.Movie;

public interface MovieDetailsContract {

    interface Model {

        interface OnFinishedListener {
            void onFinished(Movie movie);

            void onFailure(Throwable t);
        }

        void getMovieDetails(OnFinishedListener onFinishedListener, int movieId);
    }

    interface View {

        void setDataToViews(Movie movie);

        void onResponseFailure(Throwable throwable);
    }

    interface Presenter {
        void onDestroy();

        void requestMovieData(int movieId);
    }
}

File: app/src/main/java/com/example/moviemvp/movie_detail/MovieDetailsContract.kt
package com.example.moviemvp.movie_detail

import com.example.moviemvp.model.Movie

interface MovieDetailsContract {

    interface Model {

        interface OnFinishedListener {
            fun onFinished(movie: Movie)

            fun onFailure(t: Throwable)
        }

        fun getMovieDetails(onFinishedListener: OnFinishedListener, movieId: Int)
    }

    interface View {

        fun setDataToViews(movie: Movie)

        fun onResponseFailure(throwable: Throwable)
    }

    interface Presenter {
        fun onDestroy()

        fun requestMovieData(movieId: Int)
    }
}